kafka-streams-framework/src/test/java/org/hypertrace/core/kafkastreams/framework/SampleAsyncApp.java

```diff
@@ -95,3 +100,11 @@ public void process(Record<String, String> record) {
     log.info("received - key: {}, value: {}", record.key(), record.value());
   }
 }
+
+class StringHashCodePartitioner implements KeyToAsyncThreadPartitioner<String> {
```

Comment:
we can make this a first class as part of streams library as this is the most common usecase